### PR TITLE
EVG-2333: use anser sessions in migrations

### PR DESCRIPTION
**Pull request by tychoish**

### CODE REVIEW

File: migrations/testresults.go [hunk 1]
```diff
@@ -159,13 +163,13 @@ func testResultsGeneratorFactory(env anser.Environment, db string, limit int) []
 }
 
 func registerTestResultsMigrationOperations(env anser.Environment) error {
-	if err := env.RegisterManualMigrationOperation("tasks_testresults_noexecution", makeLegacyTaskMigrationFunction()); err != nil {
+	if err := env.RegisterManualMigrationOperation("tasks_testresults_noexecution", makeLegacyTaskMigrationFunction(db, tasksCollection)); err != nil {
```
Comment (bsamek) [inline migrations/testresults.go]: `db` has to come from somewhere

**Comment by tychoish** [comment]
Updated!

File: migrations/anser.go [hunk 2]
```diff
@@ -58,7 +58,7 @@ func (opts Options) Application(env anser.Environment) (*anser.Application, erro
 		Limit: opts.Limit,
 	}
 
-	if err := registerTestResultsMigrationOperations(env); err != nil {
+	if err := registerTestResultsMigrationOperations(env, opts.Database); err != nil {
```
Comment (bsamek) [inline migrations/anser.go]: Where is `opts.Database` set?

File: migrations/anser_test.go [hunk 3]
```diff
@@ -24,7 +24,10 @@ func TestAnserBasicPlaceholder(t *testing.T) {
 	assert.NoError(err)
 	assert.False(app.DryRun)
 
-	env, err := opts.Setup(context.Background())
+	ctx, cancel := context.WithCancel(context.Background())
```
Comment (bsamek) [inline migrations/anser_test.go]: Also need to fix testresults_test.go.

**Pull request merged**